IPIP/0342-content-router-discovery.md

```diff
@@ -32,6 +32,20 @@ is also insufficient long term because:
 
 ## Detailed design
 
+This spec is designed for the ability of IPFS nodes to automatically discover
+and make use of 'content routers'. Content routers are services which are able
+to fulfill libp2p's [ContentRouting](https://github.com/libp2p/go-libp2p/blob/master/core/routing/routing.go#L26)
+API. These routers currently are considered to directly support queries using
+the protocols specified by
+[IPIP-337](https://github.com/ipfs/specs/pulls)
+and/or
+[IPIP-327](https://github.com/ipfs/specs/pull/327).
```

Comment:
We need to either decide which one is the future, or document how client decides which one to use for sending requests.
IF we do the latter, including content type in response is the way to go: Reframe endpoint is  `application/vnd.ipfs.rpc[..]; version=n`.